src/data/dem_tree.js

```diff
@@ -308,6 +310,7 @@ export default class DemMinMaxQuadTree {
     }
 
     _addNode(min: number, max: number, leaf: number) {
+        this.minimum = Math.min(this.minimum, min);
```

Comment:
Leftover?

Minimum among _visibleDemTiles is reduced at getMinElevationBelowMSL - this seems to track minimum among all tiles loaded.

Reply:
Not a leftover, this minimum is for one DEM only, after going over the mips. In getMinElevationBelowMSL we then reduce it over the visible dems.

Reply:
@karimnaaji 
Thanks. It is then redundant calculation since the minimum of a DEM tile is already available at DemMinMaxQuadTree.minimums[0].

e.g. this never asserts:

```
diff --git a/src/terrain/terrain.js b/src/terrain/terrain.js
index c74f6535e..b2f1beff7 100644
--- a/src/terrain/terrain.js
+++ b/src/terrain/terrain.js
@@ -747,6 +747,7 @@ export class Terrain extends Elevation {
         const maxDEMError = 30.0;
         this._visibleDemTiles.filter(tile => tile.dem).forEach(tile => {
             const minMaxTree = (tile.dem: any).tree;
+            assert(minMaxTree.minimum == minMaxTree.minimums[0])
             min = Math.min(min, minMaxTree.minimum);
         });
         return min === 0.0 ? min : (min - maxDEMError) * this._exaggeration;

```

Similar (max for tile too) reduce is done in getBounds3D where minimums[0] is used for the same purpose.

https://github.com/mapbox/mapbox-gl-js/blob/74afcb202bbd059b00d3c931f0da886998f52cb7/src/geo/transform.js#L996

Reply:
Good catch, I'll update for minimums[0]